streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -111,6 +111,11 @@ public int compare(TopicPartition p1, TopicPartition p2) {
     public void configure(Map<String, ?> configs) {
         numStandbyReplicas = (Integer) configs.get(StreamsConfig.NUM_STANDBY_REPLICAS_CONFIG);
 
+        final String upgradeMode = (String) configs.get(StreamsConfig.UPGRADE_FROM_CONFIG);
+        if (StreamsConfig.UPGRADE_FROM_0100.equals(upgradeMode)) {
+            log.debug("Downgrading metadata version from 2 to 1 for upgrade from 0.10.0.x.");
```

Comment:
nit: I'd suggest make it INFO, as it should be rare but important for trouble shooting.